### PR TITLE
Replace key_name with ssh_provision_key_name

### PR DESCRIPTION
**Pull request by klewis0928**
##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New config Pull Request
- New role Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by fridim — CHANGES_REQUESTED** [review]
`ssh_provision_key_name` should be left empty.

If defined, you're in the case where you want to provide your own SSH keys
see https://redhat-cop.github.io/agnosticd/#_bring_your_own_provision_key

File: ansible/configs/ocp-workshop/README.adoc [hunk 1]
```diff
@@ -34,7 +34,7 @@ ansible-playbook ansible/main.yml \
   -e "cloud_provider=${CLOUDPROVIDER}" \
   -e "aws_region=${REGION}" \
   -e "HostedZoneId=${HOSTZONEID}" \
-  -e "key_name=${KEYNAME}" \
+  -e "ssh_provision_key_name=${KEYNAME}" \
```
Comment (fridim) [inline ansible/configs/ocp-workshop/README.adoc:37]: we should remove entirely that i think.
The name has a default, if you provide it, you're in the case "i already have a key, i provide it".

File: ansible/configs/ocp-workshop/README.adoc [hunk 2]
```diff
@@ -96,7 +96,7 @@ ansible-playbook ${DEPLOYER_REPO_PATH}/main.yml \
   -e "guid=${GUID}" \
   -e "env_type=${ENVTYPE}" \
   -e "cloud_provider=${CLOUDPROVIDER}" -e "aws_region=${REGION}" \
-  -e "HostedZoneId=${HOSTZONEID}" -e "key_name=${KEYNAME}" \
+  -e "HostedZoneId=${HOSTZONEID}" -e "ssh_provision_key_name=${KEYNAME}" \
```
Comment (fridim) [inline ansible/configs/ocp-workshop/README.adoc:99]: same here

File: ansible/configs/ocp-workshop/env_vars.yml [hunk 3]
```diff
@@ -95,7 +95,7 @@ docker_device: /dev/xvdb
 # set `set_env_authorized_key` to true and set the keyname in `env_authorized_key`
 # you can use the key used to create the environment or use your own self generated key
 env_authorized_key: "{{guid}}key"
-ansible_ssh_private_key_file: ~/.ssh/{{key_name}}.pem
+ansible_ssh_private_key_file: ~/.ssh/{{ssh_provision_key_name}}.pem
```
Comment (fridim) [inline ansible/configs/ocp-workshop/env_vars.yml:98]: same here

File: ansible/configs/ocp-workshop/files/cloud_providers/ec2_cloud_template.j2 [hunk 4]
```diff
@@ -191,7 +191,7 @@ Resources:
         - "RHEL74GOLD"
 {% endif %}
       InstanceType: "{{instance['flavor'][cloud_provider]}}"
-      KeyName: "{{instance['key_name'] | default(key_name)}}"
+      KeyName: "{{instance['ssh_provision_key_name'] | default(ssh_provision_key_name)}}"
```
Comment (fridim) [inline ansible/configs/ocp-workshop/files/cloud_providers/ec2_cloud_template.j2:194]:
```yaml
      KeyName: "{{ ssh_provision_key_name | default(instance['key_name']) | default(key_name)}}"
```

**Review by fridim — DISMISSED** [review]

**Review by bbethell-1 — APPROVED** [review]

**Pull request merged**